api/types/src/derives.rs

```diff
@@ -240,6 +240,36 @@ impl_poem_type!(
     )
 );
 
+impl_poem_type!(
+    U16,
+    "string",
+    (
+        example = Some(serde_json::Value::String("65534".to_string())),
+        format = Some("uint64"),
+        description = Some(indoc! {"
+      A string containing a 16-bit unsigned integer.
+
+      We represent u16 values as a string to ensure compatibility with languages such
+      as JavaScript that do not parse u64s in JSON natively.
+  "})
+    )
+);
+
+impl_poem_type!(
+    U32,
+    "string",
+    (
+        example = Some(serde_json::Value::String("2147483647".to_string())),
+        format = Some("uint64"),
+        description = Some(indoc! {"
+    A string containing a 32-bit unsigned integer.
+
+    We represent u32 values as a string to ensure compatibility with languages such
+    as JavaScript that do not parse u64s in JSON natively.
```

Comment:
copy paste typo. Should be "JavaScript that do not parse **u32s** in JSON natively"